src/core_plugins/tile_map/public/editors/wms_options.js

```diff
@@ -2,10 +2,53 @@ import { uiModules } from 'ui/modules';
 import wmsOptionsTemplate from './wms_options.html';
 const module = uiModules.get('kibana');
 
-module.directive('wmsOptions', function () {
+module.directive('wmsOptions', function (serviceSettings) {
   return {
     restrict: 'E',
     template: wmsOptionsTemplate,
     replace: true,
+    scope: {
+      options: '='
+    },
+    link: function ($scope) {
+
+      console.log('call linker wms options');
+      //todo: just noodling, not sure if this is the right way
+      $scope.options.baseLayersAreLoaded = new Promise((resolve, reject) => {
+
+        serviceSettings
+          .getTMSServices()
+          .then((allTMSServices) => {
+
+            if (!$scope.options.tmsLayers) {
+              $scope.options.tmsLayers = [];
+            }
+
+            //todo: clean up duplication with similar logic in region_map_vis_params
+            const newBaseLayers = $scope.options.tmsLayers.slice();
+            for (let i = 0; i < allTMSServices.length; i += 1) {
+              const layerFromService = allTMSServices[i];
+              const alreadyAdded = newBaseLayers.some((layer) => layerFromService.id === layer.id);
+              if (!alreadyAdded) {
+                newBaseLayers.push(layerFromService);
+              }
+            }
+            $scope.options.tmsLayers = newBaseLayers;
+
+            if (!$scope.options.selectedTmsLayer) {
+              $scope.options.selectedTmsLayer = $scope.options.tmsLayers[0];
+            }
+            resolve(true);
```

Comment:
I guess the `true` isn't needed here?

Reply:
I preferred the promise to resolve to a success-flag iso. of just undefined, as it's being used as a boolean check by the client.

Reply:
I am fine with leaving it in, even though I think, the promise should just reject if there are cases where it's not successful, and a resolved promise should always mean successful, which would render the `true` useless imho. But I won't nitpick on this one :D

Reply:
i agree 100% that if promise was not succesful it should reject instead of resolve with false